Marlin/src/HAL/HAL_STM32F1/timers.h

```diff
@@ -54,7 +55,13 @@ typedef uint16_t hal_timer_t;
 #define TEMP_TIMER_NUM 2    // index of timer to use for temperature
 //#define TEMP_TIMER_NUM 4  // 2->4, Timer 2 for Stepper Current PWM
 #define PULSE_TIMER_NUM STEP_TIMER_NUM
-#define SERVO0_TIMER_NUM 1  // SERVO0 or BLTOUCH
+
+#if MB(BIGTREE_SKR_MINI_E3) || MB(BIGTREE_SKR_E3_DIP) || MB(BTT_SKR_MINI_E3_V1_2) || MB(MKS_ROBIN_LITE)
```

Comment:
may be replaced by #if STM32_HAVE_TIMER(8)

only the CB/C8 card need timer 1...  STM32F103CB_malyan

Reply:
I did not do this, because other boards have FAN pins on timer 8. It would trade a fix on the SKR E3 boards for potential problems on other boards.

Reply:
but also locks 2 timers in Marlin for the same feature...

Reply:
Let me see if I can solve the problem by switching to different channels, rather than a different timer.

Reply:
I just meant the list of boards could take 5 pages in a few months, using this kind of code :p better to do the reverse... if the timer 8 doesnt exist...

Reply:
I was concerned about adding board details into the HAL. I don't think it is a good long-term fix.
Right now, I don't plan to make any further changes until other devs are back and are able to comment with their opinions.

Reply:
I was trying to create a larger fix that would allow the "special" boards to override timers in their pins file, but @Roxy-3D recommended this simple fix for right now.

Reply:
yep, for the mks robin, its the reverse, fan pin is TIM8_CH1...

Reply:
Hmm...perhaps an option would be to allow a pins file to provide an array of timers that are safe for software-only use. Then the HAL can index into the array, rather than assuming particular timers are safe.

Reply:
i coded it like that first, someone has put it then in this file.... was intended to be set in the pins file if used

Reply:
something to try:
#ifndef SERVO0_TIMER_NUM 
  #define SERVO0_TIMER_NUM 1
#endif

and define 8 in your board file

Reply:
That is the first thing I tried, and it did not work. This file ends up included before the pins file. That would work if I moved this #DEFINE into Servo.cpp.

Reply:
It seemed like there was some intention to have the timer numbers listed all in timers.h. If that is not true, then moving it to Servos.cpp and adding the #ifndef would be the easy fix.

Reply:
maybe let a comment in the .h ... #14407 seems related